libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -431,5 +442,10 @@ private bool ChannelSupportsOAuthCard(string channelId)
 
             return true;
         }
+
+        private bool RequiresLink(ITurnContext turnContext)
+        {
+            return !turnContext.Activity.ServiceUrl.Contains("http");
```

Comment:
Bug alert: Note that you are using Contains instead of StartsWith which is a bug. StartsWith catch things like "http://myservice.com", but since you used contains, I could have something like "ws://myhttptest" and pass as http, despite being websockets.

Reply:
I changed it to StartsWith. Thank you.